### ISSUE TITLE
Building when the game is paused option

### ISSUE THREAD

**Issue by Razip** [OpenLoco/OpenLoco#125]
I just know that some people including me would want that feature.

This feature can be off by default though.

**Comment by jgottula** [comment]
This is something I've done manually quite a bit: I'll constantly have my finger on the Pause key on the keyboard while building, so that each time the game unpauses because I built a track segment, I can immediately re-pause it, and therefore get something approaching build-while-paused. I like to be very methodical when building tracks, and it irritates me and makes me feel rushed when the game keeps running as I try to decide exactly how I want to continue the route.

This is apparently a desirable enough feature that someone even wrote [a utility](https://www.owenrudge.net/download/download?t=10&id=195) *specifically for the purpose of automating this*. (I couldn't ever get this particular program to work, unfortunately. I did plan to reverse engineer it to see what addresses they were poking to accomplish their functionality, but never did get around to that.)

> This feature can be off by default though.

I agree it should be off by default, yeah. This is one of those things that a decent proportion of players would probably argue you really shouldn't be able to do and gives you a bit of an unfair advantage over the AI, compared to letting time pass by as you carefully build a track.

So perhaps it should ultimately go in the "cheats" menu. (As a very mild, arguably-cheaty feature.)

**Comment by Razip** [comment]
@jgottula, AI builds things very very fast, which can be considered unfair too, so let the player decide whether it's cheaty or not by turning the option on or keeping it off.

**Comment by jgottula** [comment]
> AI builds things very very fast, which can be considered unfair too

@Razip Well, technically the speed at which the AI players construct their tracks/roads (which, admittedly, are always already completely pre-planned-out from the moment they begin "building" them) can range from very fast to really quite slow; it's directly related to that AI company owner's "Aggressiveness" characteristic. So an AI with Aggressiveness 9 will indeed build very fast; but an AI with Aggressiveness 1 builds unbelievably slow. (With a gradient in between those two extremes, of course.)

But more to the point, I was kind of trying to make a point based on the AI's advantages and disadvantages **as a whole**. Because while they *can* do certain things in ways that are superior to the human player (sometimes requiring the game to effectively cheat for them), which can include building quickly, they *also* frequently do very boneheaded things that put them at a *disadvantage* compared to an even moderately skilled human player. For example, building idiotically expensive bridge routes because they are too dumb to use suspension bridges and build a bunch of unnecessary curves at high altitude; or making their trains unnecessarily slow down by overusing turn segments and slopes and brick bridges; or similarly making their trucks unnecessarily slow down by overusing tight turns on their roads; or being too dumb to realize that their vehicles are stuck in some manner and taking forever to abandon the route and try something different; and on and on.

Basically, I was just saying that there are people who would argue that building while paused means giving yourself an unfair advantage (when compared against the overall strengths and weaknesses of the AI players), and so those people would consider it a bit of a "cheat" perhaps.

> so let the player decide whether it's cheaty or not by turning the option on or keeping it off

Oh don't get me wrong, I'm not disagreeing with you here. All I'm proposing is that it might be a good idea to simply categorize this in the "cheats" options rather than putting it somewhere else. (Check out OpenRCT2's cheats options to get an idea of what I mean.)

![buildwhilepaused](https://user-images.githubusercontent.com/822207/36369411-f6ac8046-150f-11e8-864c-0342549743c2.png)

**Comment by Razip** [comment]
@jgottula, yes, it needs to be categorized somewhere, but wherever is the feature, it will be very much appreciated! 😄

**Comment by wopian** [comment]
It's also in OpenTTD's cheat menu

**Comment by wopian** [comment]
It's also in OpenTTD's cheat menu

**Comment by wopian** [comment]
It's also in OpenTTD's cheat menu

**Comment by Razip** [comment]
@wopian, I can't see it.

![2018-02-19_142521](https://user-images.githubusercontent.com/6756114/36370484-cb366988-1580-11e8-97bf-bd862306a6c1.jpg)

**Comment by jgottula** [comment]
Incidentally, that is a very excellent shaming message for a cheats window.

**Comment by jgottula** [comment]
@wopian @Razip It seems that it was initially considered a cheat, but then at some point was recategorized as an advanced option.

https://wiki.openttd.org/Build_in_pause

**Comment by Razip** [comment]
@jgottula, so I'd stick to that myself. :P

**Comment by jgottula** [comment]
Ultimately it's just semantics; it doesn't actually change anything in any substantive way whether or not it's categorized as a cheat.
